common/primitives/src/utils.rs

```diff
@@ -133,13 +140,21 @@ mod tests {
 	}
 
 	#[test]
-	fn as_hex_option_can_serialize_nothing() {
+	fn as_hex_option_can_serialize_nothing_with_skip() {
 		let test_data = TestAsHexOption { data: None };
 		let result = serde_json::to_string(&test_data);
 		assert!(result.is_ok());
 		assert_eq!("{}", result.unwrap());
 	}
 
+	#[test]
+	fn as_hex_option_can_serialize_nothing_as_null() {
+		let test_data = TestAsHexOptionNull { data: None };
+		let result = serde_json::to_string(&test_data);
+		assert!(result.is_ok());
+		assert_eq!("{\"data\":null}", result.unwrap());
```

Comment:
Simple test for when you use as_hex_option, but don't want to skip it (aka null is ok.)